example_test.go

```diff
@@ -0,0 +1,60 @@
+package parquet_test
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"github.com/segmentio/parquet-go"
+)
+
+func Example() {
+	// parquet-go uses the same struct-tag definition style as JSON and XML
+	type Contact struct {
+		Name string `parquet:"name"`
+		// "zstd" specifies the compression for this column
+		PhoneNumber string `parquet:"phoneNumber,optional,zstd"`
+	}
+
+	type AddressBook struct {
+		Owner             string    `parquet:"owner,zstd"`
+		OwnerPhoneNumbers []string  `parquet:"ownerPhoneNumbers,gzip"`
+		Contacts          []Contact `parquet:"contacts"`
+	}
+
+	f, _ := os.Create("testdata/data.parquet")
+	writer := parquet.NewWriter(f)
+	rows := []AddressBook{
+		{Owner: "UserA", Contacts: []Contact{
+			{Name: "Alice", PhoneNumber: "+15505551234"},
+			{Name: "Bob"},
+		}},
+		// Add more rows here.
+	}
+	for _, row := range rows {
+		if err := writer.Write(row); err != nil {
+			log.Fatal(err)
+		}
+	}
+	_ = writer.Close()
+	_ = f.Close()
+
+	// Now, we can read from the file.
+	rf, _ := os.Open("testdata/data.parquet")
```

Comment:
How about using `/tmp` instead of testdata? Otherwise running this example will add a new file to the repository that will likely end up being checked in by accident.

Reply:
Ah, it does but we have the testdata directory checked in to gitignore currently. OK

Reply:
🤔 

That seems unexpected, we have plenty of files there in the repo https://github.com/segmentio/parquet-go/tree/main/testdata, that gitignore rule may not be stated properly and should be removed.

Reply:
https://github.com/segmentio/parquet-go/pull/171